### ISSUE TITLE
Wrong highlighting of code in languages (like R), where a dot (".") is simply a normal character used in function or object names

### ISSUE THREAD

**Issue by SebKrantz** [EnlighterJS/EnlighterJS#132]
Hello, it seems like EnlighterJS produces some wrong highlighting behavior for certain types of code containing a dot (".") as standard character in the code. My case involves highlighting of R code. [This post](https://www.r-bloggers.com/2020/08/introducing-collapse-advanced-and-fast-data-transformation-in-r/) provides an example. We see here in the first code box a call to the function `install.packages()`. This is the full name of the function in R, the "." is just a standard character in the name, thus the full function name should be highlighted, not just "packages". The post includes lots of other examples with this mistake. For example in the second code box right at the top there is a line `v <- iris$Sepal.Length`. This line pulls out a variable called "Sepal.Length" from a dataset called "iris" and stores it in a vector "v". Here no part of this code should be highlighted since no function is being called (functions always take the form FUN(), i.e. with opening and closing brackets following the name). Yet the term "Length" is highlighted, as if it were some sort of member function in JavaScript. 
In short: To fix this mistake, the "." should be treated as any other character.

**Comment by talgalili** [comment]
Hey @AndiDittrich - thanks.
Where can we modify/add these rules?

**Comment by AndiDittrich** [comment]
see https://github.com/EnlighterJS/EnlighterJS/blob/master/src/lang/r.js

to add full support a lot of examples and a detailed semantic  language reference is required as well as keyword lists, operators, etc..

**Comment by talgalili** [comment]
@AndiDittrich - I took a look at the files. I see many possible modifications possible.
I'd like to suggest making incremental improvements.
The one thing I propose is to just make sure all features treat  dot (".") as a regular character instead of a special character.

I'm looking at:
https://github.com/EnlighterJS/EnlighterJS/blob/master/src/lang/r.js
I see it imports definitions from:
https://github.com/EnlighterJS/EnlighterJS/blob/master/src/lang/generic.js

I suspect that in order to treat fo.fo() as a function call I need to copy and modify this:
https://github.com/EnlighterJS/EnlighterJS/blob/master/src/lang/rulesets/generic.js#L116
And change it from:

```
    fCalls: {
        regex: /\b([\w]+)\s*\(/gm,
        type: 'm0'
    },
```
To:

```
    fCalls: {
        regex: /\b([\w.]+)\s*\(/gm,
        type: 'm0'
    },
```
So it would include .

There are other changes I could do.

If I were to make these modifications, could you or someone else please review?

**Comment by AndiDittrich** [comment]
just doing a plain copy+paste of the generic support file is not the way to go... a real `R` support file should be recreated from scratch

**Comment by talgalili** [comment]
I don't have the time to resolve all the issues of the language. But I can deal with some of the big ones. Wouldn't it be better to have (say) 40% coverage (that is better than the current generic support). And then if people want to improve on it in the future - they could?

**Comment by AndiDittrich** [comment]
not really....it messes up the library with dead code..

**Comment by talgalili** [comment]
@AndiDittrich , I'm missing something.
Are you telling me that python module (for example) is perfect?
https://github.com/EnlighterJS/EnlighterJS/blob/master/src/lang/python.js
If so, I can copy paste what's in the python example and fix it to the R version (although, I don't think it would be perfect).

**Comment by AndiDittrich** [comment]
EnlighterJS is not designed for 100% syntactic+lexical coverage.... having 90%+ is sufficient for most use-cases.

R is a very simple language...it doesn't require much time to create a dedicated support file - BUT i'm not familiar with R -> i didn't have any real world examples, languages rules etc... if you want a better support i need some assistance

**Comment by talgalili** [comment]
Thanks.
Is there an easy way to setup a testing env?
(I.e.: files with some text box that if I fill it I'll see how the code
looks?)

**Comment by AndiDittrich** [comment]
see https://github.com/EnlighterJS/documentation/tree/master/development

but first of all, just start with collecting the basic language rulesets, keyword lists, examples as links
